cmd/kube-apiserver/app/testing/server_test.go

```diff
@@ -384,3 +385,65 @@ func crdExistsInDiscovery(client apiextensionsclientset.Interface, crd *apiexten
 	}
 	return false, nil
 }
+
+// TestOpenAPIPresence validates that various paths exist in the
+// complete delegation chain of the server. If the chain is not wired
+// up correctly this test should fail; this is largely a smoke test to
+// ensure that the complete delegation chain is complete.
```

Comment:
as far as I can tell, a successful hack/update-openapi-spec.sh does the same.

Reply:
Nope, it doesn't. A checked in file is no guarantee that nothing is broken. This is easily clicked away with a "Updated all and everything generated" commit in a broken PR. Tests are for verification that the code does what it does.

Reply:
I think in review process (and to the author) it should be suspecting if a whole API is removed from OpenAPI spec based on an unrelated change. Basically if you change removed apiregistration and you did not removed apiregistration endpoint either author or the reviewer should catch it and if they ignore it, I don't see it different than removing this test.

The test here hardcode a specific api group. What if we rename/replace it? Why this endpoint? why not others? The aggregator can fail in many ways and can remove any of those API groups. I just don't see the benefit of adding one single apigroup test here. The whole point of checking in OpenAPI spec was to make sure our changes would not result in unintentional changes in API surface.

Reply:
It's a smoke test that we don't break OpenAPI delegation. It's explicit, the stored openapi json is not. I have seen many many PRs where people just regenerated code (e.g. completely broken conversions) and no reviewer or approver noticed. Generated code is too often under the radar in reviews.